pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/segment/SegmentAssignmentUtils.java

```diff
@@ -408,6 +409,25 @@ public Map<String, Map<String, String>> getNonTierSegmentAssignment() {
     }
   }
 
+  /**
+   * Checks tiers one-by-one to return the first one for which a newly registered segment qualifies
+   * @param tableNameWithType table to which the segment assignment belongs
+   * @param sortedTiers list of tiers, pre-sorted as per desired order by caller
+   * @param segmentName name of the new segment
+   * @return tier for which the segment qualifies, or null if segment doesn't belong to any tier
+   */
+  public static Tier findTierForNewSegment(String tableNameWithType, @Nullable List<Tier> sortedTiers,
```

Comment:
move this to TierConfigUtils.java instead?

Reply:
Fixed.